README.md

```diff
@@ -1,3 +1,167 @@
+- `email`: Settings that are used by the EmailManager. Required.
+   - `smtp`: Nodemailer SMTP Configuration, this object will be passed directly to Nodemailer without any further processing. Check the [Nodemailer Docs](https://nodemailer.com/smtp/#general-options). If you need a service to test emails locally you can use [Ethereal](https://ethereal.email/), a dummy email service that doesn't actually sends emails but simulates a fully-featured SMTP server.
+   - `display`: Display settings how the server will refer to itself in emails.
+      - `sender`: The Entry for the `From:` field in the email, the actual email should be the correct one, otherwise the emails will likely land in SPAM Folders. Required.
+      - `server`: Settings that define how the Server will refer to itself in E-Mails. Required.
+         - `protocol`: The Protocol that should be used for this server in emails. Default `http`.
+         - `host`: The hostname that should be used to refer to this server, ideally a domain. Default `localhost`.
+         - `port`: The port that should be used to connect to this server, if used behind a reverse-proxy this should be the public port. Default: `7654`.
+         - `baseurl`: In case your server is in a non-root installation, set this to the folder name. (Example `yourserver.com/dice` -> `/dice`.) Default ` `.
+- `keys`:
+   - `private`: Path to the private key used to sign dice rolls. Required.
+   - `public`: Path to the public key used to sign dice rolls. Required.
+   
+### Starting
+In order to start the server run
+`node dice-server.js`
+It can be terminated using `SIGTERM`, i.e. `Ctrl+C`
+
+### Testing
+In order to run all local tests and eslint, you can simply run
+`yarn test`
+This is also the command that will be executed by Travis.
+
+## Routes
+The dice server is divided into 2 seperate routers.
+The REST Service handles all calls under `/api`.
+All other requests are handled by the _frontend_ which basically wraps the API calls with a nice UI.
+### API
+All of the requests return JSON in the same format.
+When successful:
+```json
+{
+  "status": "OK",
+  "result": {
+    "some info": "some value",
+    "other info": true
+  }
+}
+```
+`result` is optional, if it's not present this simply means that the server has no additional information for this request.
+On Error:
+```json
+{
+  "status": "Error",
+  "errors": ["List", "of", "error", "messages"]
+}
+```
+#### Routes
+- POST `/api/roll`:
+   - Required Parameters (POST Body, urlencoded):
+      - `max` Integer. Specifies the highest outcome for each roll, must be <= 100.
+      - `times` Integer. Specifies how many dice should be rolled, must be <= 100.
+      - `email1` String. The first email to send the notification email to. This emails needs to be registered.
+      - `email2` String. The second email to send the notification to. This email needs to be registered.
+   - Generates `times` random numbers with a value in `[1, times]`.
```

Comment:
Should the second reference to `times` actually be `max`?

```suggestion
   - Generates `times` random numbers with a value in `[1, max]`.
```

Reply:
Yes, took my way too long to write all of this (but in the end revealed some minor problems, which is a good thing), so at some point I must have lost my patience. 😅